### PR TITLE
[Fix] Add --[no-]shrink instruction for release builds

### PR DESCRIPTION
**Pull request by piedcipher**
Add --[no-]shrink instruction for release builds.

Fixes: #10032

## Presubmit checklist

- [x] This PR is marked as draft with an explanation if not meant to land until a future stable release.
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

**Review by domesticmouse — APPROVED** [review]

**Review by parlough — COMMENTED** [review]
Thanks for working to address this older issue!

It seems like instead of having a note, this section needs to be removed or rewritten to not mention `--[no-]shrink` at all anymore.

@gmackall Is there anything still relevant to developers for shrinking? Is it sufficient to just tell them that Flutter runs shrinking on release builds and link to the Android shrink docs?

File: src/content/deployment/android.md [hunk 1]
```diff
@@ -249,6 +249,8 @@ To disable R8, pass the `--no-shrink` flag to
 :::note
 Obfuscation and minification can considerably extend compile time
 of the Android application.
+
+`--[no-]shrink` flag has no effect. Code shrinking is always enabled in release builds. To learn more, see: [Shrink, obfuscate, and optimize your app](https://developer.android.com/studio/build/shrink-code)
```
Comment (sfshaza2) [inline src/content/deployment/android.md:253]:
```suggestion
The `--[no-]shrink` flag has no effect. Code shrinking is always enabled in release builds.
To learn more, check out
[Shrink, obfuscate, and optimize your app]({{site.android-dev}}/studio/build/shrink-code).
```

**Comment by gmackall** [comment]
> Thanks for working to address this older issue!
> 
> It seems like instead of having a note, this section needs to be removed or rewritten to not mention `--[no-]shrink` at all anymore.
> 
> @gmackall Is there anything still relevant to developers for shrinking? Is it sufficient to just tell them that Flutter runs shrinking on release builds and link to the Android shrink docs?

Sorry for missing this ping! Yes, that is correct. From what I can tell, the flag has been ignored since
https://github.com/flutter/flutter/commit/8dd0de7f580972079f610a56a689b0a9c414f81e#diff-3c44d2e8279ea8d917b4fb942252d4649689154c36cf74227a1a2a879aaf1915L328
Use of the Gradle property was brought back in 
https://github.com/flutter/flutter/pull/76192/files#diff-3c44d2e8279ea8d917b4fb942252d4649689154c36cf74227a1a2a879aaf1915R384
to disable shrinking when building with deferred components, but not the ability to set the property (so it is only modified internally).

So
1. The flag has no effect
2. Shrinking is always enabled on release builds, except when building with [deferred components](https://docs.flutter.dev/perf/deferred-components).

**Review by sfshaza2 — APPROVED** [review]
lgtm

**Pull request merged**